### ISSUE TITLE
[Bug]: OpenAI wrong cost calculation for cached_tokens

### ISSUE THREAD

**Issue by paulthemagno** [BerriAI/litellm#6215]
### What happened?

I'm sorry if I'm wrong, but for my understanding the cost calculation for the new `cached_tokens` of OpenAI APIs works in this way:
```json
"usage": {
    "prompt_tokens": 2006,
    "completion_tokens": 300,
    "total_tokens": 2306,
    "prompt_tokens_details": {
        "cached_tokens": 1920
    },
    "completion_tokens_details": {
        "reasoning_tokens": 0
    }
}
```

`prompt_tokens` should contain both cached and not cached tokens. So to calculate the real cost the formula should be
`(prompt_tokens - cached_tokens) * cost_not_cached_prompt_tokens + cached_tokens * cost_cached_prompt_tokens + 
completion_tokens * cost_completion_tokens`. Am I right ?

In this snippet, it seems that it multiplies the `prompt_tokens` * cost of the not cached tokens. In my opinion, it should subtract the number of `cached_tokens` before doing that, since they are considered also after in the code :
[https://github.com/BerriAI/litellm/blob/d0a30529374ac45923f0038f3556919cb50cb906/litellm/llms/OpenAI/cost_calculation.py#L29-L42](https://github.com/BerriAI/litellm/blob/d0a30529374ac45923f0038f3556919cb50cb906/litellm/llms/OpenAI/cost_calculation.py#L29-L42)

Am I making some mistakes?

**Comment by ishaan-jaff** [comment]
yes - you're right @paulthemagno, fixed here: https://github.com/BerriAI/litellm/pull/6231

**Comment by ishaan-jaff** [comment]
I checked: https://platform.openai.com/docs/api-reference/chat/object and https://community.openai.com/t/will-cached-prompt-be-charged-in-each-api-call/977999 to confirm

**Comment by ishaan-jaff** [comment]
curious do you use LiteLLM proxy in production @paulthemagno and @loretoparisi ? 

hoping to get user feedback
